src/librbd/AioObjectRequest.h

```diff
@@ -45,6 +45,10 @@ namespace librbd {
       return !m_parent_extents.empty();
     }
 
+    virtual const char *get_op_type() const = 0;
+
+    virtual bool pre_object_map_update(uint8_t *new_state) = 0;
+
```

Comment:
This is needed for CopyupRequest to handle generic object requests. We could make these private and declare CopyupRequest to be a friend if that is preferable.